### ISSUE TITLE
migrate recent changes to rsconnect

### ISSUE THREAD

**Issue by jjallaire** [rstudio/shinyapps#155]
I attempted to pull from shinyapps upstream into rsconnect and the merge has become **extremely** complicated with the most recent set of changes. *Please, please, please* make new commits into rsconnect so that we don't risk introducing defects during complex merges. 

As we begin testing rsconnect more aggressively we are finding defects and fixing them (e.g. we just found a unicode escaping issue in RCurl and fixed it). We are also changing deployment internals to handle various types of static content. We are also fixing issues with interactive documents on both the rsconnect client and server side. These changes (which are now happening at a rapid rate) all make the merges more complicated and defect prone. 

Further, all of the fixes you make to shinyapps for e.g. improved locale support, bioconductor support, etc. are not available to IDE customers who choose to deploy using the UI (which is now much more aggressively promoted in v0.99). Even customers that start with the shinyapps package are force migrated to rsconnect the first time the UI is used so while they may have for a time gotten e.g. the locale or bioconductor updates these changes are erased the first time they use the UI to deploy.

We are preparing to come out with a patch release for v0.99 of IDE which will be the first one we do global notification of all users for. It is critical that all of the recent work in shinyapps/rsconnect is included in this update.

In short, we need to pull the shinyapps changes into rsconnect *right away* and then shut down new commits to shinyapps (or as @kippandrew suggested invert the dependency to make shinyapps essentially an alias to rsconnect).

@kippandrew @CBarraford  @aronatkins  @jcheng5 @jmcphers @tareefk

**Comment by kippandrew** [comment]
Closed by #156
